apps/web/app/(use-page-wrapper)/router/page.tsx

```diff
@@ -0,0 +1,43 @@
+import type { PageProps as _PageProps } from "app/_types";
+import type { GetServerSidePropsContext } from "next";
+import { cookies, headers } from "next/headers";
+import { redirect } from "next/navigation";
+import { cache } from "react";
+
+import { getRoutedUrl } from "@calcom/lib/server";
+
+import { buildLegacyCtx } from "@lib/buildLegacyCtx";
+
+const getCachedRoutingFormData = cache(async (legacyCtx: GetServerSidePropsContext) => {
+  return await getRoutedUrl(legacyCtx);
+});
```

Comment:
Cache this call so we dont need to refetch this data in the title and loading the page